fix for 31592: check if value for divider is too big

okaufman opened the pull request:
Fixes https://mantis.ilias.de/view.php?id=31592
When trying to save a value for divider that is to big, a message will show up: 
"This number has to be smaller than the maximum range."

okaufman committed:
fix for 31592

kergomard commented:
Hi @okaufmann!
As you were the first one of the group to create a PR (so cool, thank you very much!) I had a look atit and have three little questions/comments:
- I think it would be good to change the comparison from `if ( $intPrecision >= $max_range) {` to `if ( $intPrecision->getValue() >= $max_range->getValue()) {`, right? Otherwise, I think, but really correct me if I'm wrong, you are relying on $value being the first variable in the object.
- As the range is inclusive: Wouldn't it be better to use ">" than ">="?  It seems a little random to not include the upper limit.
- Finally: This only addresses the first of the two cases (ie. it prevents errors for new questions), but it doesn't address the problem for existing questions. Is this on purpose?

Thanks again!
Stephan

mbecker-databay commented:
I think we can skip the existing questions: Due to the error they seem to be totally unusable, so not hidden from anyones sight.
But that is a decision I would pass to @dsstrassner 
Regarding the language file additions: "Nummer" -> "Number" or "Wert" -> "Value" I see there.

Great job, thank you so much!

mjansenDatabay commented on Modules/TestQuestionPool/classes/class.assFormulaQuestionGUI.php:
```diff
@@ -689,6 +689,11 @@ public function editQuestion($checkonly = false)
                         $custom_errors = true;
                     }
+                    $intPrecision = $form->getItemByPostVar('intprecision_' . $variable->getVariable());
+                    if ( $intPrecision >= $max_range) {
```
```suggestion
                    if ($intPrecision >= $max_range) {
```

okaufman committed:
fix for 31592, part 2

okaufman committed:
Fix for 31592 part 3, deals with the questions that are already saved

dsstrassner commented:
> I think we can skip the existing questions: Due to the error they seem to be totally unusable, so not hidden from anyones sight. But that is a decision I would pass to @dsstrassner Regarding the language file additions: "Nummer" -> "Number" or "Wert" -> "Value" I see there.
> 
> Great job, thank you so much!

Existing questions in the database we really do not need to touch. Super. Many thanks, @okaufman

okaufman committed:
cleanup class.assFormulaQuestionVariable.php

mjansenDatabay commented on Modules/TestQuestionPool/classes/class.assFormulaQuestionVariable.php:
```diff
@@ -47,6 +47,12 @@ public function getRandomValue()
     {
         
 //		@todo check this
+        if ($this->getIntprecision() > $this->getRangeMax())
```
This is not valid `PSR-2 + X` code style.

okaufman commented:
@swiniker  thank you for looking at my pull-request. I made the changes you suggested. 
I also added another if-statement in  getRandomValue() to check for the problem in already existing questions.

@mbecker-databay Thank you for your comment. I changed the the lang file to value/Wert

@mjansenDatabay I removed the extra space. Thank you for your sugestion.
